### PR TITLE
Cleanup build-dist-js task

### PR DESCRIPTION
**Pull request by AlanFoster**
This PR improves the `build-dist-js` gulp task by -

- Cleaning up the `dist` folder before attempting to build any files
- Ignore building any test related files

### CODE REVIEW

File: dist/utils/shallowEqual.js [hunk 1]
```diff
@@ -35,8 +35,10 @@ function shallowEqual(objA, objB) {
         if (!objA[key].isSame(objB[key], 'day')) {
           return false;
         }
-      } else if ((0, _isMomentRange2['default'])(objA[key]) && (0, _isMomentRange2['default'])(objB[key]) && !(0, _areMomentRangesEqual2['default'])(objA[key], objB[key])) {
-        return false;
+      } else if ((0, _isMomentRange2['default'])(objA[key]) && (0, _isMomentRange2['default'])(objB[key])) {
+        if (!(0, _areMomentRangesEqual2['default'])(objA[key], objB[key])) {
```
Comment (AlanFoster) [inline dist/utils/shallowEqual.js]: This file wasn't re-generated as part of #72

**Comment by jkimbo** [comment]
@AlanFoster this could be setup as a `prepublish` npm script and then the dist folder can be removed from repo completely which would make pr diffs a lot nicer? It's something that I was meaning to do but never got around to it.

Example: https://github.com/gaearon/library-boilerplate/blob/3b49912f1f6b2a3fcbfadb11921a00ad0e361e85/package.json#L14

**Comment by AlanFoster** [comment]
@jkimbo I'm all for that idea :+1:

For visibility, what is the specific release process for this component?  I think I will be releasing the next version shortly, and I don't want to miss any steps. The long term plan is to automate releases with gulp, but I'd like for the release steps to be documented initially.

**Comment by jkimbo** [comment]
The process I followed was just bumping the npm version and then after publishing running `gulp deploy-example` which should handle pushing things to the `gh-pages` branch.

**Pull request merged**